tests/unity_perception/test_download.py

```diff
@@ -33,19 +33,41 @@ def downloader():
 
 
 @responses.activate
-def test_download_file_from_url():
+def test_download_file_from_url_with_filename():
     source_uri = "https://mock.uri"
     body = b"some test string here"
     responses.add(
         responses.GET, source_uri, body=body, content_type="text/plain"
     )
 
     with tempfile.TemporaryDirectory() as tmp_dir:
-        dest_path = os.path.join(tmp_dir, "test.txt")
-        download_file(source_uri, dest_path, file_name="file.txt")
+        file_path = download_file(source_uri, tmp_dir, file_name="file.txt")
```

Comment:
I actually have second thought about testing this file_path here. I think a robust test should include:

1. assert the `file_path` return by download_file match the expected path, `tmp_dir/file.txt`.
2. assert `tmp_dir/file.txt` exists
3. assert the content of `tmp_dir/file.txt` matches `body`

Reply:
Done.